src/Query/LaravelReadQuery.php

```diff
@@ -105,19 +105,22 @@ public function getResourceSet(
             assert($numValues == count($segments));
 
             for ($i = 0; $i < $numValues; $i++) {
-                $relation = $segments[$i]->isAscending() ? '>=' : '<=';
+                $relation = $segments[$i]->isAscending() ? '>' : '<';
                 $name = $segments[$i]->getSubPathSegments()[0]->getName();
-                $line = ['direction' => $relation, 'value' => $values[$i][0]];
-                $parameters[$name] = $line;
+                $parameters[$name] = ['direction' => $relation, 'value' => trim($values[$i][0], "\'")];
             }
 
             foreach ($parameters as $name => $line) {
-                $direction = $line['direction'];
-                if ($keyName == $name) {
-                    $direction = '!=';
-                }
-                $value = trim($line['value'], "\'");
-                $sourceEntityInstance = $sourceEntityInstance->where($name, $direction, $value);
+                $processed[$name] = ['direction' => $line['direction'], 'value' => $line['value']];
```

Comment:
This is where the O ( N ** 2 ) nature of skipTokenry hits as outlined above - once we've added a property as where greater than, we then drop it back to where equal to